Source/EasyNetQ/Consumer/ConsumerDispatcher.cs

```diff
@@ -8,53 +8,73 @@ namespace EasyNetQ.Consumer
     public class ConsumerDispatcher : IConsumerDispatcher
     {
         private readonly ILog logger = LogProvider.For<ConsumerDispatcher>();
-        private readonly BlockingCollection<Action> queue;
-        private bool disposed;
+        private readonly AutoResetEvent autoResetEvent = new AutoResetEvent(false);
+        private readonly ConcurrentQueue<Action> durableActions = new ConcurrentQueue<Action>();
+        private readonly ConcurrentQueue<Action> transientActions = new ConcurrentQueue<Action>();
 
         public ConsumerDispatcher(ConnectionConfiguration configuration)
         {
             Preconditions.CheckNotNull(configuration, "configuration");
 
-            queue = new BlockingCollection<Action>();
-
             var thread = new Thread(_ =>
             {
-                while (!disposed && queue.TryTake(out var action, -1))
-                {
```

Comment:
I think it is better not to remove `{` `}` for multiline blocks.

Reply:
I say get rid of all the {}'s :)

Reply:
I do not recommend to do this for multiline blocks.

Reply:
What to do guys  ? :) i am in between :D i really think it depends on the case, and it in this case it could be both :)

Reply:
Let's discuss it [here](https://github.com/EasyNetQ/EasyNetQ/pull/1026), please. Leave the braces as it was, please, just to simplify merging.

Reply:
Is it fine @Rotvig?